portality/scripts/anon_import.py

```diff
@@ -0,0 +1,94 @@
+    if config.get("elastic_search_db") is not None:
+        index = config.get("elastic_search_db")
+        app.config["ELASTIC_SEARCH_DB"] = index
+
+    print("\n")
+    print("Using host {x} and index {y}\n".format(x=host, y=index))
+    conn = esprit.raw.make_connection(None, host, None, index)
+
+    # filter for the types we are going to work with
+    import_types = {}
+    for t, s in config.get("types", {}).iteritems():
+        if s.get("import", False) is True:
+            import_types[t] = s
+
+    print("==Carrying out the following import==")
+    for import_type, cfg in import_types.iteritems():
+        count = "All" if cfg.get("limit", -1) == -1 else cfg.get("limit")
+        print("{x} from {y}".format(x=count, y=import_type))
+    print("\n")
+
+    if config.get("confirm", True):
+        text = raw_input("Continue? [y/n]")
+        if text.lower() != "y":
+            exit()
+
+    # remove all the types that we are going to import
+    for import_type in import_types.keys():
+        esprit.raw.delete(conn, import_type)
+
+    # re-initialise the index (sorting out mappings, etc)
+    print("==Initialising Index for Mappings==")
+    initialise_index(app)
+
+    mainStore = StoreFactory.get("anon_data")
+    tempStore = StoreFactory.tmp()
+    container = constants.STORE_ANON_DATA
+
+    print("\n==Importing==")
+    for import_type, cfg in import_types.iteritems():
+        count = "all" if cfg.get("limit", -1) == -1 else cfg.get("limit")
+        print("Importing {x} from {y}".format(x=count, y=import_type))
+        print("Obtaining {x} from storage".format(x=import_type))
+
+        limit = cfg.get("limit", -1)
+        limit = None if limit == -1 else limit
+
+        n = 1
+        while True:
+            filename = import_type + ".bulk" + "." + str(n)
+            handle = mainStore.get(container, filename)
+            if handle is None:
+                break
+            print("Retrieved {x} from storage".format(x=filename))
+            tempStore.store(container, filename, source_stream=handle)
+            handle.close()
+
+            data_file = tempStore.path(container, filename)
+            imported_count = esprit.tasks.bulk_load(conn, import_type, data_file,
+                                                    limit=limit, max_content_length=config.get("max_content_length", 100000000))
+            tempStore.delete(container, filename)
+
+            if limit is not None and imported_count != -1:
+                limit -= imported_count
+            if limit <= 0:
+                break
+
+            n += 1
+
+    tempStore.delete(container)
+
+if __name__ == '__main__':
+
+    import argparse
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument("config", help="Config file for import run")
+    args = parser.parse_args()
+
+    with codecs.open(args.config, "rb", "utf-8") as f:
+        config = json.loads(f.read())
+
+    do_import(config)
```

Comment:
Can ```config``` be undefined here?

Reply:
no.  config is a positional command line argument, so if it is omitted argparse will give you an error